source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+         hpc-worker2: UpTime=0-01:34:56
+         hpc-worker1: NodeName=hpc-worker1 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=721
+         hpc-worker1: UpTime=0-01:39:21
+
+   #. Get their IP addresses.
+
+      .. code-block:: bash
+
+         pdsh -w ^worker-nodes-list "networkctl status | awk 'FNR == 2 {print $2}'"
+
+      Example output:
+
+      .. code-block:: console
+
+         hpc-worker4:          Address: 192.168.1.23 on eth0
+         hpc-worker3:          Address: 192.168.1.22 on eth0
+         hpc-worker2:          Address: 192.168.1.21 on eth0
+         hpc-worker1:          Address: 192.168.1.20 on eth0
+
+   #. :command:`sudoedit` the :file:`slurm.conf` file.  Append each worker node 
+      information plus its IP address, but without the `UpTime`, under the 
+      `COMPUTE NODES` section.
+
+      .. tip::
+
+         It is strongly recommended to set the `RealMemory` value for each 
+	 worker node slightly below, say 90%, what was reported by 
+	 :command:`slurmd -C` 
+	 in case some memory gets use by some processes, which would 
+	 cause slurm not to make a node not available due to its memory 
+	 resource falling below the stated value in the configuration file.    
+
+      Here's an example with four worker nodes:
+
+      .. code-block:: console
+
+         # 
+         # COMPUTE NODES (mode detailed values reported by "slurmd -C" on each node)
+         NodeName=hpc-worker1 NodeAddr=192.168.1.20 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=648
+         NodeName=hpc-worker2 NodeAddr=192.168.1.21 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=648
+         NodeName=hpc-worker3 NodeAddr=192.168.1.22 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1497
+         NodeName=hpc-worker4 NodeAddr=192.168.1.23 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1723
+
+   #. Create partitions.
+
+      A slurm partition is basically the grouping of worker nodes.  
+      Give each partition a name and decide which worker node(s) go belong to 
+      it.
+
+      For example:
+   
+      .. code-block:: console
+      
+         PartitionName=workers Nodes=hpc-worker1,hpc-worker2,hpc-worker3, hpc-worker4 Default=YES MaxTime=INFINITE State=UP
+         PartitionName=debug Nodes=hpc-worker1,hpc-worker3 MaxTime=INFINITE State=UP
+
+         PartitionName=workers Nodes=hpc-worker[1-4] Default=YES MaxTime=INFINITE State=UP
+         PartitionName=debug Nodes=hpc-worker[1,3] MaxTime=INFINITE State=UP
+
+   #. Save and exit.
+
+   #. Set the ownership of the :file:`slurm.conf` file to slurm.
+
+      .. code-block:: bash
+         
+	 sudo chown slurm: /etc/slurm/slurm.conf
+
+Copy :file:`slurm.conf` and munge key to all worker nodes
+*********************************************************
+
```

Comment:
I am pretty sure you need to start munge service before slurmd service on each node. So the order should be reversed. If I'm wrong then I just learned something new. Please see : https://slurm.schedmd.com/quickstart_admin.html 
Quoting:
 Make sure that all nodes in your cluster have the same munge.key. Make sure the MUNGE daemon, munged is started before you start the Slurm daemons.

Reply:
It seemed to work the way I had it.  But I'll take the advice of starting MUNGE before Slurm daemons.